test-repos/nodeJS-v2/functions.js

```diff
@@ -0,0 +1,79 @@
+module.exports = {
+  goodbye: {
+    handler: 'handler.hello',
+    events: [
+      {
+        http: {
+          path: 'goodbye',
+          method: 'get',
+          cors: true,
+          responseData: {
+            200: {
+              description: 'this went well',
+              bodyType: 'goodbye',
+            },
+          },
+          queryStringParameters: {
+            who: {
+              required: true,
+              type: 'string',
+              description: 'the person to say goodbye to',
+            },
+            count: {
+              required: false,
+              type: 'integer',
+            },
+          },
+        },
+      },
+    ],
+  },
+
+  hello: {
+    handler: 'handler.hello',
+    events: [
+      {
+        http: {
+          path: 'hello/{userID}',
+          method: 'get',
+          cors: true,
+        },
+      },
+    ],
+  },
+  hellopost: {
+    handler: 'handler.hello',
+    events: [
+      {
+        http: {
+          path: 'hello',
+          method: 'post',
+          cors: true,
+          bodyType: 'helloPostBody',
+          summary: 'This is a test of a post request to do something',
+          headerParameters: {
+            authorisation: {
+              required: true,
+              description: 'the authorisation token',
+            },
```

Comment:
Same here, should cover the other branches of implicit and explicit required === false
```suggestion
            },
            explicitOptional: {
              required: false,
              description: 'this header is optional',
            },
            implicitOptional: {
              description: 'this header is implicitly optional',
            },
```